controller/ingress_entry.go

```diff
@@ -6,9 +6,24 @@ import (
 	"time"
 )
 
-// IngressUpdate data
-type IngressUpdate struct {
-	Entries []IngressEntry
+// IngressEntries type
+type IngressEntries []IngressEntry
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Len() int {
+	return len(entries)
+}
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Less(i, j int) bool {
+	iEntry := fmt.Sprintf("%s:%s:%s:%s", entries[i].Namespace, entries[i].Name, entries[i].Host, entries[i].Path)
+	jEntry := fmt.Sprintf("%s:%s:%s:%s", entries[j].Namespace, entries[j].Name, entries[j].Host, entries[j].Path)
```

Comment:
I put this here because it's where IngressUpdate was previously defined, and it's also where the IngressEntry struct is being defined.